### PR TITLE
[data] enable test_split.py tests

### PR DESCRIPTION
**Pull request by matthewdeng**
Signed-off-by: Matthew Deng <matt@anyscale.com>

## Why are these changes needed?

This test was being skipped in CI because it was not running pytest in the main script (https://github.com/ray-project/ray/pull/25322)

**Changes:**
1. Add `if __name__ == "__main__":` condition to run the test.
2. Updated `test_train_test_split` results based on random seed.
    1. I believe the original value was never correct, and reproduced the updated results on 2 different environments.
3. Disabled `split(equal=False)` logic in `test_equal_split`.
    1. This seems to be a regression since 1.13.0, though I'm not certain what the expected behavior is around how "even" the splits should be. In 1.13.0 the splits were [7, 7, 6] and [8, 6, 6], now the splits are [10, 5, 5] and [8, 8, 4].

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by scv119 — APPROVED** [review]

**Comment by scv119** [comment]
@matthewdeng rebase master?

**Review by c21 — APPROVED** [review]
